orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -693,31 +720,30 @@ Status TrainingRunner::RunWithoutUpdate(VectorString& feed_names,
   // Async launch of a session.
   pipeline_worker_pool_.workers[worker_id] = std::thread([&](
       const size_t worker_id, const size_t step) {
-#if !defined(NDEBUG) && defined(USE_CUDA) && !defined(_WIN32)
+#ifdef ENABLE_NVTX_PROFILE
     // Store the tag for the thread which runs session_.Run(...).
     // It will be used to name range in Nvidia's visual profiler.
     auto& profile_context = profile::Context::GetInstance();
     profile_context.SetThreadTag(
       std::this_thread::get_id(), std::to_string(step));
+#else
+    ORT_UNUSED_PARAMETER(step);
 #endif
-    // Dummy use of step to avoid warning when the code above is disabled. 
-    ORT_ENFORCE(step + 1 > 0);
     RunOptions run_options;
     run_options.only_execute_path_to_fetches = true;
-    ORT_ENFORCE(session_.Run(
+    auto status = session_.Run(
       run_options,
       pipeline_worker_pool_.worker_states[worker_id].feed_names,
       pipeline_worker_pool_.worker_states[worker_id].feeds,
       pipeline_worker_pool_.worker_states[worker_id].fetch_names,
-      &(pipeline_worker_pool_.worker_states[worker_id].fetches)) == Status::OK());
+      &(pipeline_worker_pool_.worker_states[worker_id].fetches));
+    ORT_THROW_IF_ERROR(status);
```

Comment:
do we want to throw from within the thread?

Reply:
perhaps not, uncaught exceptions in a std::thread seem to call terminate()
https://en.cppreference.com/w/cpp/thread/thread

Reply:
it's ok with me to address it in a separate PR though, this issue was there before